modules/sampler_hijack.py

```diff
@@ -234,21 +234,28 @@ def __call__(self, input_ids: torch.LongTensor, scores: torch.FloatTensor) -> to
             input_ids = input_ids[:, -self._range:]
 
         for input_ids_row, scores_row in zip(input_ids, scores):
+
+            # Use normal Python data types for improved performance
+            s = input_ids_row.tolist()
+            
             # Raw integer must be extracted here to check for set membership.
-            last_token = input_ids_row[-1].item()
+            last_token = s[-1]
 
             if last_token in self.sequence_breakers:
                 continue
 
             # Exclude the last token as it always matches.
-            match_indices = (input_ids_row[:-1] == last_token).nonzero()
+            match_indices = []
+            for idx, val in enumerate(s[:-1]):
+                if val == last_token:
+                    match_indices.append(idx)
```

Comment:
Here's an idea: What about instead of converting the tensor to a Python list, you convert it to a `np.array` (NumPy is already a dependency)?

This would give you access to `np.argwhere` etc., which might be quite a bit faster than a Python loop, while *also* being simpler and cleaner. And I doubt that NumPy has the same indexing overhead that Torch seems to have.

Reply:
It seems that there are many different ways to iterate over elements in a NumPy array, and all of them are slower than iterating elements on a Python list: https://github.com/numpy/numpy/issues/16985

Reply:
Thanks for the info. Guess I have to revisit my simplistic assumption "if you want speed, use NumPy". Pretty crazy that this hasn't been optimized yet.